File: leetcode012/main.py
def hammingWeight( n: int) -> int:
    res = 0
    while n : # while n != 0
        res += n % 2
        n = n >> 1
    return res


if __name__ == "__main__":
    num = 123
    res = hammingWeight(num)
    print(res)

File: leetcode012/main.go
package main

import "fmt"

func main() {
	res := hammingWeight(1111111111)
	fmt.Println(res)

	var t uint32 = 2
	fmt.Println(t % 2)
}

func hammingWeight(num uint32) int {
	var bits int
	for num != 0 {
		num = num & (num - 1)
		bits += 1
	}
	return bits
}
func hammingWeight2(num uint32) int {
	var bits uint32

	for num != 0 {
		bits += num % 2
		num = num >> 1
	}

	return int(bits)
}
